### PR TITLE
Do operating system deployment in migration test cases

### PR DESCRIPTION
**Pull request by neo954**
Enhance existing migration test cases.
This pull request is for task #2856

### CODE REVIEW

**Comment by hu-weihua** [comment]
@neo954 , I did not get the usage of ``go-xcat smoke-test`` in ``go-xcat`` usage information. But anyway, I don't think your change has achieved what we expected. I suggest we should do below steps:

* install lower version xcat
* Do basic definitions base on lower version xcat, such like makedns, makedhcp, node definition and osimage definition.
* upgrade to higher version xcat
* Do provision to verify all definitions still work in new version xcat

File: xCAT-test/autotest/testcase/migration/redhat_migration [hunk 1]
```diff
@@ -93,12 +95,14 @@ check:rc==0
...
 check:rc==0
 check:output=~running
-cmd:xdsh $$CN "lsdef"
-check:output=~node0001
-cmd:xdsh $$CN "noderm node0001"
-check:rc==0
 cmd:latest_version_info=`lsxcatd -v`;xdsh $$CN "lsxcatd -v" | grep "$latest_version_info"
 check:rc==0
+cmd:ssh $$CN "nodeset $$SN osimage=__GETNODEATTR($$CN,os)__-__GETNODEATTR($$CN,arch)__-install-compute"
+cmd:rpower $$SN boot
```
Comment (hu-weihua) [inline xCAT-test/autotest/testcase/migration/redhat_migration]: Maybe use ``rinstall`` is better option, this case need to be run on LPAR.

File: xCAT-test/autotest/testcase/migration/redhat_migration [hunk 2]
```diff
@@ -93,12 +95,14 @@ check:rc==0
...
+check:rc==0
+cmd:ssh $$CN "a=0; while ! `lsdef -l $$SN grep status|grep booted >/dev/null`; do ((++a > 90)) && break; sleep 20; done"
+cmd:ssh $$CN "xdsh $$SN date"
+check:rc==0
```
Comment (hu-weihua) [inline xCAT-test/autotest/testcase/migration/redhat_migration]: In fact, we did not check if the SN was installed successfully. right? If SN was not installed successfully,but enter initrd,  when line 103 is time out, line 104 still can pass.  So after line 103, we still need to check the ``booted`` again.

**Comment by neo954** [comment]
@hu-weihua, Test cases updated based on your comments.

**Comment by hu-weihua** [comment]
It is good for me, I agree to merge. thanks @neo954

**Pull request merged**